wurst/_items/BattleglovesCritical.wurst

```diff
@@ -0,0 +1,114 @@
+package BattleglovesCritical
+
+import AbilityObjEditing
+import ObjectIdGenerator
+import ObjectIds
+import HashMap
+import ClosureEvents
+import ClosureTimers
+import DamageType
+import ID
+
+let CRITICAL_ABILITY_ID = compiletime(ABIL_ID_GEN.next())
+let SPELLBOOK_ABILITY_ID = compiletime(ABIL_ID_GEN.next())
+let COOLDOWN = 6.
+let COOLDOWN_SKIP_PER_ATTACK = 1.
+let MULTIPLIER = 1.6
+let CLOCK_TICK_INTERVAL = 0.05
+
+@compiletime function createCriticalAbility()
+    new AbilityDefinitionBladeMasterCriticalStrike(CRITICAL_ABILITY_ID)
+    ..setItemAbility(false)
+    ..setHeroAbility(false)
+    ..presetDamageMultiplier(lvl -> MULTIPLIER)
+    ..presetDamageBonus(lvl -> 0)
+    ..presetChancetoCriticalStrike(lvl -> 100)
+    ..presetChancetoEvade(lvl -> 0)
+    ..presetCooldown(lvl -> 0)
+    ..presetNeverMiss(lvl -> true)
+
+@compiletime function createHiddenSpellbook()
```

Comment:
Could you create a compiletime function that takes in a list of ability IDs and creates a spellbook containing those IDs. That way we could reuse this for other hidden abilities.

Reply:
Created